File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cs
public class Solution {
    public string ReverseVowels(string s) {
        char[] word = s.ToCharArray();
        
        int p1 = 0;
        int p2 = word.Length - 1;

        while(p1 < p2) {
            while(p1 < p2 && !IsVowel(word[p1])) {
                p1++;
            }
            while(p1 < p2 && !IsVowel(word[p2])) {
                p2--;
            }

            (word[p1], word[p2]) = (word[p2], word[p1]);
                
            p1++;
            p2--;
        }

        return new string(word);

        bool IsVowel(char c) {
            return c is 'a' or 'e' or 'i' or 'o' or 'u'
                or 'A' or 'E' or 'I' or 'O' or 'U';
        }
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int n = s.size();
        int i=0;
        int j=n-1;
        string vowels = "aeiouAEIOU";


        while(i<j){
            while(i<j && vowels.find(s[i])==string::npos) i++;
            while(i<j && vowels.find(s[j])==string::npos) j--;
            swap(s[i],s[j]);
            i++;
            j--;
        }
        return s;
    }
};